cmake/CoverallsGenerateGcov.cmake

```diff
@@ -184,30 +184,29 @@ message("GCDA files:")
 # Get a list of all the object directories needed by gcov
 # (The directories the .gcda files and .o files are found in)
 # and run gcov on those.
-foreach(GCDA ${GCDA_FILES})
-	message("Process: ${GCDA}")
-	message("------------------------------------------------------------------------------")
-	get_filename_component(GCDA_DIR ${GCDA} PATH)
+message("Processing GCDA files")
+message("------------------------------------------------------------------------------")
+list(GET GCDA_FILES 0 FIRST_GCDA)
+get_filename_component(GCDA_DIR ${FIRST_GCDA} PATH)
 
-	#
-	# The -p below refers to "Preserve path components",
-	# This means that the generated gcov filename of a source file will
-	# keep the original files entire filepath, but / is replaced with #.
-	# Example:
-	#
-	# /path/to/project/root/build/CMakeFiles/the_file.dir/subdir/the_file.c.gcda
-	# ------------------------------------------------------------------------------
-	# File '/path/to/project/root/subdir/the_file.c'
-	# Lines executed:68.34% of 199
-	# /path/to/project/root/subdir/the_file.c:creating '#path#to#project#root#subdir#the_file.c.gcov'
-	#
-	# If -p is not specified then the file is named only "the_file.c.gcov"
-	#
-	execute_process(
-		COMMAND ${GCOV_EXECUTABLE} -p -o ${GCDA_DIR} ${GCDA}
-		WORKING_DIRECTORY ${COV_PATH}
-	)
-endforeach()
+#
+# The -p below refers to "Preserve path components",
+# This means that the generated gcov filename of a source file will
+# keep the original files entire filepath, but / is replaced with #.
+# Example:
+#
+# /path/to/project/root/build/CMakeFiles/the_file.dir/subdir/the_file.c.gcda
+# ------------------------------------------------------------------------------
+# File '/path/to/project/root/subdir/the_file.c'
+# Lines executed:68.34% of 199
+# /path/to/project/root/subdir/the_file.c:creating '#path#to#project#root#subdir#the_file.c.gcov'
+#
+# If -p is not specified then the file is named only "the_file.c.gcov"
+#
+execute_process(
+	COMMAND ${GCOV_EXECUTABLE} -p -o ${GCDA_DIR} ${GCDA_FILES}
```

Comment:
Hmm, I just realized that this will not work if you generate GCDAs in serveral folders.
Previously, each gcda file got its proper GCDA_DIR as argument here, but now the first one is used for all 🙄

But gcov refuses multiple -o arguments, so I don't see how this can possibly solved without resorting to a more powerful tool, i.e. lcov.

Reply:
Hmm well, can't really merge this like it is, since it will break stuff...

But if you mean lcov can do the exact same thing but better,  using that here might be an option